solidity/test/FullyBackedECDSAKeepFactoryTest.js

```diff
@@ -0,0 +1,1685 @@
+      // Ethereum uint256 division performs implicit floor:
+      // The weight went down from 200 to 199
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns false if the operator bonding value increased", async () => {
+      const unbondedValue = minimumBondableValue.muln(10) // (20 * 10^18) * 10
+      await setUnbondedValue(members[0], unbondedValue)
+
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], unbondedValue.add(precision)) // ((20 * 10^18) * 10) + 10^18
+
+      // Precision (bond weight divisor) is 10^18
+      // (20 * 10^18 * 10) / 10^18           = 200
+      // ((20 * 10^18 * 10) + 10^18) / 10^18 = 201
+      // The weight went up from 200 to 201
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns true if the operator bonding value increase is below weight precision", async () => {
+      const unbondedValue = minimumBondableValue.muln(10) // (20 * 10^18) * 10
+      await setUnbondedValue(members[0], unbondedValue)
+
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], unbondedValue.add(precision).subn(1)) // ((20 * 10^18) * 10) + 10^18 - 1
+
+      // Precision (pool weight divisor) is 10^18
+      // (20 * 10^18 * 10) / 10^18                 =  200
+      // ((20 * 10^18 * 10) + (10^18 - 1)) / 10^18 =~ 200.99
+      // Ethereum uint256 division performs implicit floor:
+      // The weight dit not change: 200 == floor(200.99)
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns false if the operator bonding value dropped below minimum", async () => {
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], minimumBondableValue.subn(1))
+
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("reverts if the operator is not registered for the application", async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+  })
+
+  describe("updateOperatorStatus", async () => {
+    let minimumBondableValue
+
+    before(async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+      await registerMemberCandidates()
+
+      const pool = await FullyBackedSortitionPool.at(signerPool)
+      minimumBondableValue = await pool.getMinimumBondableValue()
+
+      await setUnbondedValue(members[0], minimumBondableValue.muln(3))
```

Comment:
This call happens after `joinPool` executed inside `registerMemberCandidates` so all tests are starting with the operator being not up-to-date. That's why `revers if operator is up to date` is failing. We should probably move this call inside `registerMemberCandidates` or call an update here.